### PR TITLE
feat(nx-cloud): new cloud onboarding flow

### PR DESCRIPTION
**Pull request by mandarini**
I closed the previous 2 PRs, and just keeping this last one that contains all.

~~THIS PR IS THE LAST IN A CHAIN OF 3 PRs~~ 
~~The first PR of the chain is https://github.com/nrwl/nx/pull/26007~~
~~The second PR in the chain is: https://github.com/nrwl/nx/pull/26243~~
~~So, if this PR is ok, then we can only just merge this one.~~

Shorten URLs before showing to the user.

Fallback to full URL if shortening fails.

(bonus PR added to the chain: https://github.com/nrwl/nx/pull/26290)

Screenshot 2024-05-30 at 5 29 43 PM

Screenshot 2024-05-30 at 5 30 18 PM

Screenshot 2024-05-30 at 5 31 26 PM

### CODE REVIEW

File: packages/nx/src/command-line/connect/connect-to-nx-cloud.ts [hunk 1]
```diff
@@ -51,22 +52,43 @@ export async function connectToNxCloudIfExplicitlyAsked(
...
+      });
+    } else {
+      const token =
+        process.env.NX_CLOUD_ACCESS_TOKEN || nxJson.nxCloudAccessToken;
+      if (!token) {
+        throw new Error(
+          `Unable to authenticate. Either define accessToken in nx.json or set the NX_CLOUD_ACCESS_TOKEN env variable.`
+        );
+      }
+      const connectCloudUrl = await shortenedCloudUrl('nx-connect', token);
```
[the remark this answers is not in the document]
Comment (mandarini) [reply packages/nx/src/command-line/connect/connect-to-nx-cloud.ts:73]: Or, do you mean we should send a request to ocean to verify that token is valid before creating the URL with the token?

[the remark this answers is not in the document]
Comment (jaysoo) [reply packages/nx/src/command-line/connect/connect-to-nx-cloud.ts:73]: Ah okay, that's fine. I think invalid tokens would be handled by App either when shortening or later in the flow.

File: packages/nx/src/nx-cloud/utilities/url-shorten.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (!response?.data || response.data.message) {
+      throw new Error(
+        response?.data?.message ?? 'Failed to shorten Nx Cloud URL'
+      );
+    }
+
+    return `${apiUrl}/connect/${response.data}`;
+  } catch (e) {
+    logger.warn(`Failed to shorten Nx Cloud URL.
```
Comment (jaysoo) [inline packages/nx/src/nx-cloud/utilities/url-shorten.ts:46]: I think we should not the user the error here. The error on the App side should already be reported, and it's a bit jarring to see the error given that they aren't blocked from continuing the flow.

Maybe stick it into `logger.verbose` for us to debug if needed.

Comment (mandarini) [reply packages/nx/src/nx-cloud/utilities/url-shorten.ts:46]: Ok!

**Review by jaysoo — APPROVED** [review]

**Review by JamesHenry — APPROVED** [review]

**Pull request merged**